processor/metricstransformprocessor/README.md

```diff
@@ -0,0 +1,116 @@
+# Metrics Transform Processor <span style="color:red">**(UNDER DEVELOPMENT - NOT READY FOR USE)**</span>
+- <span style="color:red">This ONLY supports renames/aggregations **within individual metric batches**.</span> It does not do any aggregation across batches, so it is not suitable for aggregating metrics from multiple sources (e.g. multiple nodes or clients). At this point, it is only for aggregating metrics from a single source that groups its metrics for a particular time period into a single batch (e.g. host metrics from the VM the collector is running on).
+- <span style="color:red">Rename Collisions will result in a no operation on the metrics data</span>
+  - e.g. If want to rename a metric or label to `new_name` while there is already a metric or label called `new_name`, this operation will not take any effect. There will also be an error logged
+
+## Description
+The objective of this metrics transform processor is to give OpenTelemetry Collector users the flexibility to rename and aggregate metrics in desired ways so that the metrics are more relevant and cheaper to the users. This processor can be used to simply rename metrics names, labels and label values to conform with any requirements of an existing backend. This processor can also perform aggreagtions on metrics across labels or label values, so that the data can cost less and provide more insights for specific use cases.
```

Comment:
```suggestion
The metrics transform processor can be used to rename metrics names, labels and/or label values, and to perform aggregations on metrics across labels and/or label values.
```

I don't think you need to justify the need for this component in the README. It's better to just focus on what it does and how to use it imo.